admiral/pkg/clusters/serviceentry.go

```diff
@@ -189,19 +190,98 @@ func createServiceEntryForNewServiceOrPod(env string, sourceIdentity string, rem
 			//} else {
 			//	log.Errorf(LogErrFormat, "Create", istioModel.VirtualService.Type, virtualServiceName, rc.ClusterID, err)
 			//}
+			if common.GetWorkloadSidecarUpdates() == "enabled" {
+				for _, val := range dependents.Map() {
+					remoteRegistry.AdmiralCache.DependencyNamespaceCache.Put(val, serviceInstance.Namespace, localFqdn)
+				}
+
+				for _, sidecarEgress := range remoteRegistry.AdmiralCache.DependencyNamespaceCache.Get(sourceIdentity) {
+					modifySidecarForLocalClusterCommunication(serviceInstance.Namespace, sidecarEgress.Namespace, sidecarEgress.FQDN, rc)
+				}
+			}
 		}
 	}
 	return serviceEntries
 }
 
+func modifySidecarForLocalClusterCommunication(sidecarNamespace string, dependencyNamespace string, localFqdn string, rc *RemoteController) {
+
+	//get existing sidecar from the cluster
+	sidecarConfig := rc.SidecarController
+
+	if sidecarConfig == nil {
+		return
+	}
+	// sidecarConfig := rc.IstioConfigStore.Get(istioModel.Sidecar.Type, SidecarName, sidecarNamespace)
+
+	//sidecar := &networking.Sidecar{}
+
+	sidecar, _ := sidecarConfig.IstioClient.NetworkingV1alpha3().Sidecars(sidecarNamespace).Get(SidecarName, v12.GetOptions{})
```

Comment:
If sidecar is `nil` it should return from the method. We should only modify a sidecar if one exists. Cannot really create one and start restricting since that might break other functionality on the workload.